File: BalancedBrackets/BalancedBrackets.js
function balancedBrackets(str) {
  let line = str.split('')
  const stack = []
  const openers = {
    '(': ')',
    '{': '}',
    '[': ']',
  }

  const closers = {
    ')': true,
    '}': true,
    ']': true
  }

  for (let i = 0; i < line.length; i++) {
    if (openers[line[i]]) {
      stack.push(line[i])
    } else if (closers[line[i]]) {
      if (openers[stack.pop()] !== line[i]) return false
    }
  }
  return stack.length === 0
}

console.log(balancedBrackets('{}[]()'));        // should print true
console.log(balancedBrackets('{(([]))}'));      // should print true
console.log(balancedBrackets('{ [ ] ( ) }'));   // should print true
console.log(balancedBrackets('{ [ ( ] ) }'));   // should print false
console.log(balancedBrackets('('));             // should print false
console.log(balancedBrackets('{[}'));           // should print false

File: BalancedBrackets/Balanced_Brackets.py
def balancedBrackets(str):
    line = list(str)
    stack = []
    openers = {
        '(': ')',
        '{': '}',
        '[': ']',
    }

    closers = {
        ")": True,
        "}": True,
        "]": True,
    }

    for c in line:
        if c in openers:
            stack.append(c)
        elif c in closers:
            if (openers[stack.pop()] != c):
                return False
    return len(stack) == 0


print(balancedBrackets('{}[]()'))        # should print True
print(balancedBrackets('{(([]))}'))      # should print True
print(balancedBrackets('{ [ ] ( ) }'))   # should print True
print(balancedBrackets('{ [ ( ] ) }'))   # should print False
print(balancedBrackets('('))             # should print False
print(balancedBrackets('{[}'))           # should print False
